readme.md

```diff
@@ -1,161 +1,112 @@
-# AsyncAPI To-Go! :package: 
+- [Overview](#overview)
+- [Technical requirements](#technical-requirements)
+- [Supported protocols](#supported-protocols)
+- [How to use the template](#how-to-use-the-template)
+  * [CLI](#cli)
+- [Template configuration](#template-configuration)
 
-Do you bang your head against the wall writing event driven microservices?  
-Well now you you can bang your head faster!
 
-To-Go helps produce the code behind your event-driven operations by transforming AsyncAPI specs **To Go**
+## Overview
 
-## What is it?
-AsyncAPI To-Go is a code generation template for the [AsyncAPI generator](https://github.com/asyncapi/generator) that can be used to generate boilerplate Go code for interacting with event-driven architectures. 
+This template generates a Go module that uses [watermill](https://github.com/ThreeDotsLabs/watermill) as the messaging middleware
 
-The generator consumes an [**AsyncAPI 2.0**](https://www.asyncapi.com/docs/specifications/2.0.0/) YAML or JSON specification file and uses this template to produce matching channel constants, message payload structs, and (W.I.P.) ~~optionally, logic to wrap it into a lightweight library.~~ The underlying transport layer interface enables pluggable support for most messaging system (Kafka, MQTT, AMPQ, etc.).
+## Technical requirements
 
-The template provides switches to allow for different levels of generation. More information can be found at [generation levels](#generation-levels)
-* Channel parsing / generation with parameter structs
-* Schema and Message definitions
-* ~~Content Type parsing with the ability to extend supported content types~~ (Soon)
-* A transport interface to support various underlying event messaging protocols
-* ~~A simple wrapping framework to provide AsyncAPI *operations* as high-level functions~~ (Very Soon)
+- 1.1.0 =< [Generator](https://github.com/asyncapi/generator/) < 2.0.0,
+- Generator specific [requirements](https://github.com/asyncapi/generator/#requirements)
 
-## What Am I Getting Myself Into? 
-Code Samples:
+## Supported protocols
 
-Channel Building / Parsing
-```go
-p := &channel.TurnOnParams{"<my-id>"}
-p.Build() // Returns: "smartylighting/streetlights/1/0/action/<my-id>/turn/on"
+Currently this template supports AMQP subscribers
 
-p := &channel.ReceiveLightMeasurementParams{}
-err := p.Parse("smartylighting/streetlights/1/0/event/(.+)/lighting/measured")
-// Populates p with channel parameters
-// p = { StreetlightId: "<my-id>" }
-```
+## How to use the template
+
+This template must be used with the AsyncAPI Generator. You can find all available options [here](https://github.com/asyncapi/generator/).
+
+### CLI
+
+As of this initial commit this template has been tested to generate an AMQP subscriber for the asyncapi.yaml file located in the `test` folder
```

Comment:
```suggestion
This template has been tested to generate an AMQP subscriber for [this asyncapi.yaml file](./test/asyncapi.yaml).
```

how about this? we link to the file and also do not refer to a commit but the template in generate, at state when people read the readme file